src/core/relaxation_scheme.jl

```diff
@@ -1,12 +1,46 @@
+
+"""
+Computes the valid domain of a given JuMP varibale taking into account bounds
+and the varaible's implicit bounds (e.g. binary).
+"""
+function variable_domain(var::JuMP.VariableRef)
```

Comment:
This doesn't check for semi-infinite and semi-continuous sets. However, it's not trivial to detect if these constraints exist before JuMP 0.20. Maybe worth a note/todo.

Reply:
Will do.  Thanks for the review!